docs/ldml/tr35-keyboards.md

```diff
@@ -1076,435 +1294,783 @@ On some architectures, applications may directly interact with keys before they
+    The input context has:
 
-_Attribute:_ `error="fail"` (optional)
+    - …
+    - marker `circ_marker`
 
-> If set this attribute indicates that the keyboarding application may indicate an error to the user in some way. Processing may stop and rewind to any state before the key was pressed. If processing does stop, no further transforms on the same input are applied. The `@error` attribute takes the value `"fail"`, or must be absent. If processing continues, the `@to` is used for output as normal. It thus should contain a reasonable value.
+3. Also due to the `<display>` element, implementations can opt to display a visible `^` (perhaps visually distinct from a plain `^` carat). Implementations may opt to display nothing and only store the marker in the input context.
 
-For example:
+4. The user now presses the `e` key, which is also added to the input context. The input context now has:
 
-```xml
-<transform from="\u037A\u037A" to="\u037A" error="fail" />
-```
+    - …
+    - character `e`
+    - marker `circ_marker`
 
-This indicates that it is an error to type two iota subscripts immediately after each other.
+5. Now, the input context matches the transform.  The `e` and the marker are replaced with `ê`.
 
-In terms of how these different attributes work in processing a sequences of transforms, consider the transform:
+    The input context now has:
 
-```xml
-<transform before="X" from="Y" after="Z" to="B" />
-```
+    - …
+    - character `ê`
 
-This would transform the string:
+**Effect of markers on final text**
 
-```
-XYZ → XBZ
-```
+All markers must be removed before text is returned to the application from the input context.
+If the input context changes, such as if the cursor or mouse moves the insertion point somewhere else, all markers in the input context are removed.
 
-If we mark where the current match position is before and after the transform we see:
+**Implementation Notes**
 
-```
-X | Y Z → X B | Z
-```
+Ideally, markers are implemented entirely out-of-band from the normal text stream. However, implementations _may_ choose to map each marker to a [Unicode private-use character](https://www.unicode.org/glossary/#private_use_character) for use only within the implementation’s processing and temporary storage in the input context.
+
+For example, the first marker encountered could be represented as U+E000, the second by U+E001 and so on.  If a regex processing engine were used, then those PUA characters could be processed through the existing regex processing engine.  `[^\uE000-\uE009]` could be used as an expression to match a character that is not a marker, and `[Ee]\u{E000}` could match `E` or `e` followed by the first marker.
```

Comment:
Given our recent discussions on markers in Keyman, I wonder if this advice may be unhelpful/misleading and perhaps we should just leave it to the implementers to figure out how they want to solve? PUA characters could be used by an experimental keyboard for a new orthography, even if that keyboard is not acceptable for publication into CLDR.

Reply:
```suggestion
Markers should be implemented out-of-band from the normal text stream. Implementations should take care that all normal Unicode codepoints, including private-use area (PUA), are not affected by any scheme for encoding markers in text.
```

Reply:
This was implemented when this PR merged.